src/test/regress/expected/distributed_functions.out

```diff
@@ -829,6 +829,202 @@ SELECT * FROM test ORDER BY id;
 (2 rows)
 
 DROP TABLE test;
+-- verify that recreating distributed functions with TABLE params gets propagated to workers
```

Comment:
It seems like Postgres distinguishes functions/procedures/aggregates (and also routines, but not sure what is a routine? Probably just a placeholder name for the other 3.)

See here: https://github.com/postgres/postgres/blob/65c6cab1365ac33b11a49a2a193f6b3f9c53e487/src/backend/parser/parse_func.c#L2207

Should we expand our test matrix to include all the combinations -- as much as syntax allows? 
[function, procedure, aggregate, routine] -> [in param, variadic param, out param, inout param, table out param, record param out] ->  [superuser owned, non-superuser owned, owned by regular(or super) user but granted EXECUTE to another user]

We are going to backport this, and I really don't want to break any users :)

Reply:
Current tests:
Function with RETURN TABLE
Function with OUT param
Function with INOUT param
Function with VARIADIC param
Function returning SETOF int
Function returning SETOF int + with VARIADIC param
Procedure with VARIADIC param
Procedure with INOUT param

Reply:
Great!